src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -600,6 +601,9 @@ private void InitializeCustomNodeManager()
                     {
                         customNodeSearchRegistry.Remove(info.FunctionId);
                         SearchModel.Remove(searchElement);
+                        var workspacesToRemove = _workspaces.FindAll(w => w is CustomNodeWorkspaceModel
+                            && (w as CustomNodeWorkspaceModel).CustomNodeId == id);
+                        workspacesToRemove.ForEach(w => RemoveWorkspace(w));
                     }
```

Comment:
`_workspaces.RemoveAll` will just remove models from the list. `RemoveWorkspace` besides removing will trigger a corresponding event (subscribers will perform some necessary actions) and make disposing correctly.
So I'd like to leave it as it is